sql/moz-fx-data-marketing-prod/ga_derived/www_site_hits_v2/query.sql

```diff
@@ -0,0 +1,294 @@
+  WHERE
+    _TABLE_SUFFIX = FORMAT_DATE('%Y%m%d', @submission_date)
+  GROUP BY
+    1,
+    2,
+    3
+),
+get_all_events_in_each_session_staging AS (
+  SELECT
+    PARSE_DATE('%Y%m%d', a.event_date) AS date,
+    a.user_pseudo_id || '-' || CAST(
+      (
+        SELECT
+          `value`
+        FROM
+          UNNEST(event_params)
+        WHERE
+          key = 'ga_session_id'
+        LIMIT
+          1
+      ).int_value AS string
+    ) AS visit_identifier,
+    a.user_pseudo_id AS full_visitor_id,
+    a.event_name,
+    a.event_timestamp,
+    device.category AS device_category,
+    device.operating_system AS operating_system,
+    device.language AS language,
+    device.web_info.browser AS browser,
+    device.web_info.browser_version AS browser_version,
+    geo.country AS country,
+    collected_traffic_source.manual_source AS source,
+    collected_traffic_source.manual_medium AS medium,
+    collected_traffic_source.manual_campaign_name AS campaign,
+    collected_traffic_source.manual_content AS ad_content,
+    (
+      SELECT
+        `value`
+      FROM
+        UNNEST(event_params)
+      WHERE
+        key = 'engagement_time_msec'
+      LIMIT
+        1
+    ).int_value AS engagement_time_msec, --not sure if this is the same as hit time yet or not
+    (
+      SELECT
+        `value`
+      FROM
+        UNNEST(event_params)
+      WHERE
+        key = 'engaged_session_event'
+      LIMIT
+        1
+    ).int_value AS engaged_session_event,
+    (
+      SELECT
+        `value`
+      FROM
+        UNNEST(event_params)
+      WHERE
+        key = 'page_location'
+      LIMIT
+        1
+    ).string_value AS page_location,
+    COALESCE(
+      (SELECT `value` FROM UNNEST(event_params) WHERE key = 'entrances' LIMIT 1).int_value,
+      0
+    ) AS is_entrance
+  FROM
+    `moz-fx-data-marketing-prod.analytics_313696158.events_*` a
+  WHERE
+    _TABLE_SUFFIX = FORMAT_DATE('%Y%m%d', @submission_date)
+),
+--if there are multiple events at the same timestamp in the same session, assign them all the same hit number
+get_all_events_in_each_session AS (
+  SELECT
+    a.date,
+    a.visit_identifier,
+    a.full_visitor_id,
+    a.event_name,
+    a.event_timestamp,
+    a.device_category,
+    a.operating_system,
+    a.language,
+    a.browser,
+    a.browser_version,
+    a.country,
+    a.source,
+    a.medium,
+    a.campaign,
+    a.ad_content,
+    a.engagement_time_msec,
+    a.engaged_session_event,
+    SPLIT(a.page_location, '?')[OFFSET(0)] AS page_location,
+    a.is_entrance,
+    DENSE_RANK() OVER (PARTITION BY visit_identifier ORDER BY event_timestamp ASC) AS hit_number,
+    ROW_NUMBER() OVER (PARTITION BY visit_identifier ORDER BY event_timestamp) AS row_nbr
+  FROM
+    get_all_events_in_each_session_staging a
+),
+--get the hit number associated with the last page view in each session
+hit_nbr_of_last_page_view_in_each_session AS (
+  SELECT
+    visit_identifier,
+    COUNT(1) AS nbr_page_view_events,
```

Comment:
hint: Could use COUNT(*). Nbr is confusing, I'd suggest number. You decide,

```suggestion
    COUNT(*) AS number_page_view_events,
```